Invalid assert in TestDaemonStartWithoutColors integration test

**Description**

TestDaemonStartWithoutColors test for "--raw-logs=true" option will always get empty output, but it can pass the assert:

  `c.Assert(b.String(), check.Not(checker.Contains), infoLog)`

This is because "tty" is closed in s.d.Stop() for "--raw-logs=false" test, so the "--raw-logs=true" test will receive empty output.

**Steps to reproduce the issue:**
1. add following assert to TestDaemonStartWithoutColors
```
--- a/integration-cli/docker_cli_daemon_test.go
+++ b/integration-cli/docker_cli_daemon_test.go
@@ -2194,6 +2194,7 @@ func (s *DockerDaemonSuite) TestDaemonStartWithoutColors(c *check.C) {
        // Disable coloring explicitly
        s.d.StartWithLogFile(tty, "--raw-logs=true")
        s.d.Stop(c)
+       c.Assert(b.String(), check.Not(check.Equals), "")
        c.Assert(b.String(), check.Not(checker.Contains), infoLog)
 }
```
2. run integration test

`TESTFLAGS='-check.f TestDaemonStartWithoutColors' make test-integration-cli`

**Describe the results you received:**

```
----------------------------------------------------------------------
FAIL: docker_cli_daemon_test.go:2172: DockerDaemonSuite.TestDaemonStartWithoutColors

[dab45615d06d6] waiting for daemon to start
[dab45615d06d6] daemon started

[dab45615d06d6] exiting daemon
[dab45615d06d6] waiting for daemon to start
[dab45615d06d6] daemon started

[dab45615d06d6] exiting daemon
docker_cli_daemon_test.go:2197:
    c.Assert(b.String(), check.Not(check.Equals), "")
... obtained string = ""
... expected string = ""

OOPS: 0 passed, 1 FAILED
--- FAIL: Test (15.07s)
FAIL
```

**Describe the results you expected:**

Test should passed.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.4
 Git commit:   a2b12b5
 Built:        Thu Dec 29 08:20:27 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.4
 Git commit:   a2b12b5
 Built:        Thu Dec 29 08:20:27 2016
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

Not required.

**Additional environment details (AWS, VirtualBox, physical, etc.):**

--- a/integration-cli/docker_cli_daemon_test.go
+++ b/integration-cli/docker_cli_daemon_test.go
@@ -2165,6 +2165,9 @@ func (s *DockerDaemonSuite) TestDaemonStartWithoutColors(c *check.C) {
 
 	infoLog := "\x1b[34mINFO\x1b"
 
+	b := bytes.NewBuffer(nil)
+	done := make(chan bool)
+
 	p, tty, err := pty.Open()
 	c.Assert(err, checker.IsNil)
 	defer func() {
@@ -2172,19 +2175,38 @@ func (s *DockerDaemonSuite) TestDaemonStartWithoutColors(c *check.C) {
 		p.Close()
 	}()
 
-	b := bytes.NewBuffer(nil)
-	go io.Copy(b, p)
+	go func() {
+		io.Copy(b, p)
+		done <- true
+	}()
 
 	// Enable coloring explicitly
 	s.d.StartWithLogFile(tty, "--raw-logs=false")
 	s.d.Stop(c)
+	// Wait for io.Copy() before checking output
+	<-done
 	c.Assert(b.String(), checker.Contains, infoLog)
 
 	b.Reset()
 
+	// "tty" is already closed in prev s.d.Stop(),
+	// we have to close the other side "p" and open another pair of
+	// pty for the next test.
+	p.Close()
+	p, tty, err = pty.Open()
+	c.Assert(err, checker.IsNil)
+
+	go func() {
+		io.Copy(b, p)
+		done <- true
+	}()
+
 	// Disable coloring explicitly
 	s.d.StartWithLogFile(tty, "--raw-logs=true")
 	s.d.Stop(c)
+	// Wait for io.Copy() before checking output
+	<-done
+	c.Assert(b.String(), check.Not(check.Equals), "")
 	c.Assert(b.String(), check.Not(checker.Contains), infoLog)
 }
 

